### PR TITLE
Forecast refactor

### PR DESCRIPTION
**Pull request by allieIsOkay**
- Adds service classes for geocoding and forecast request to remove code from route.
- Moves services off HTTPS library and on to node-fetch
- Adds facade for forecast endpoint to further reduce code present in route.
- Adds authenticate method to user model for verifying users via API key.

@aprildagonese could you please take a look at these refactors and let me know if you see any place that I could further refactor the forecast endpoint? Additionally, if you see any places where I may have missed an edge case or could use some better documentation.

### CODE REVIEW

File: facades/forecast.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var User = require('../models').User;
+var GeocodeService = require('../services/geocode');
+var ForecastService = require('../services/forecast');
+
+module.exports = class ForecastFacade {
+  constructor(status, body) {
```
Comment (aprildagonese) [inline facades/forecast.js:8]: This is the first file I'm looking at, and I'm going to keep digging in, but off the top I'm not super clear on what "status" would be for a ForecastFacade. It's possible that you've already found the most descriptive term there, but if not, is there anything more descriptive you could use?

File: facades/forecast.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      User.authenticate(api_key)
+      .then(function() {
+        return GeocodeService.requestLocation(location)
+      })
+      .then(function(location) {
+        console.log(location);
+        return ForecastService.requestForecast(location)
+      })
+      .then(function(forecast) {
+        delete forecast.minutely;
```
Comment (aprildagonese) [inline facades/forecast.js:23]: @plapicola why did you need to delete the minutely? Did you choose that over creating logic in the forecast for whether or not you needed the minutely?

File: routes/api/v1/forecast.js [hunk 3]
```diff
@@ -1,72 +1,14 @@
 var express = require('express');
 var router = express.Router();
-var User = require('../../../models').User;
-require('dotenv').config(); // Loads environment variables from .env file
-const https = require('https');
```
Comment (aprildagonese) [inline routes/api/v1/forecast.js:5]: @plapicola why the choice to pull this out? Did you end up moving it somewhere else?

File: facades/forecast.js (hunk 1 above)
Comment (aprildagonese) [reply facades/forecast.js:8]: Ah, the request status, I get it now.

**Review by aprildagonese — APPROVED** [review]
This is beautiful! It looks a lot cleaner than your first iteration. It took me a minute to figure out what you were doing sometimes, which might just be helped by making sure your terms are descriptive, but I thought it was fairly readable still. You could even just consider adding comments if you'd prefer not to change your terms. Seems like JS is more forgiving of comments than Ruby :)

File: facades/forecast.js (hunk 2 above)
Comment (allieIsOkay) [reply facades/forecast.js:23]: I still need to format the request to more closely match the spec, but for an initial pass just opted to delete the minutely information as this was not requested in the spec. There are additional nested keys I'll need to work over to resolve. A separate class might be helpful in achieving this. I'd love your thoughts.

File: routes/api/v1/forecast.js (hunk 3 above)
Comment (allieIsOkay) [reply routes/api/v1/forecast.js:5]: I wound up moving the user down to the facade, the dotenv files to the service, and in the case of https, ditched it entirely in favor of node-fetch.

**Comment by allieIsOkay** [comment]
@aprildagonese Added some documentation to the facade, do you think it's helpful?
